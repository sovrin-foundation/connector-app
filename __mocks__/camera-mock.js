// @flow
import React from 'react'
import { View } from 'react-native'

const constants = {
  Aspect: {},
  BarCodeType: {},
  Type: {},
  CaptureMode: {},
  CaptureTarget: {},
  CaptureQuality: {},
  Orientation: {},
  FlashMode: {},
  TorchMode: {},
}

// We don't need to type check external library
// and we don't need types for mock test stub
class MockCamera extends React.Component<any, void> {
  static constants = constants
  static checkVideoAuthorizationStatus = jest.fn(() => Promise.resolve(true))

  render() {
    return <View>{this.props.children}</View>
  }
}

MockCamera.constants = constants

export default MockCamera
