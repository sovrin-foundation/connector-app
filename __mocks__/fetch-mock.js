import { Headers, Response, Request } from 'node-fetch'

const fetch = jest.fn(() => {})
fetch.Headers = Headers
fetch.Request = Request
fetch.Response = Response

fetch.mockResponse = (body, init) =>
  fetch.mockImplementation(
    () =>
      new Promise((resolve, reject) =>
        process.nextTick(() => resolve(new Response(body, init)))
      )
  )

fetch.mockResponseOnce = (body, init) =>
  fetch.mockImplementationOnce(
    () =>
      new Promise((resolve, reject) =>
        process.nextTick(() => resolve(new Response(body, init)))
      )
  )

fetch.mockResponses = (...responses) =>
  responses.map(([body, init]) =>
    fetch.mockImplementationOnce(
      () =>
        new Promise((resolve, reject) =>
          process.nextTick(() => resolve(new Response(body, init)))
        )
    )
  )

export default fetch
