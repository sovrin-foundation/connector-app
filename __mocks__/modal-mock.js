// @flow
import React from 'react'

const MockModal = (props: any) =>
  React.createElement('Modal', props, props.children)

export default MockModal
