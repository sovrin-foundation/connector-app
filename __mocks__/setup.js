import React from 'react'
import { NativeModules } from 'react-native'
import fetch from './fetch-mock'
import mockCamera from './camera-mock'
import mockModal from './modal-mock'
import mockView from './view-mock'

// mock Math.random for <Loader />
const mockMath = Object.create(global.Math)
mockMath.random = () => 0.1
global.Math = mockMath

jest.mock('NativeAnimatedHelper')

// mock this module to allow react-navigation to mock Linking
jest.mock('Linking', () => ({
  addEventListener: jest.fn(),
  removeEventListener: jest.fn(),
  openURL: jest.fn(() => Promise.resolve()),
  canOpenURL: jest.fn(() => Promise.resolve()),
  getInitialURL: jest.fn(() => Promise.resolve()),
}))

jest.mock('react-navigation', () => ({
  createStackNavigator: jest.fn(() => mockView),
  createTabNavigator: jest.fn(() => mockView),
  TabBarTop: jest.fn(),
  isFocused: true,
  withNavigationFocus: jest.fn(() => mockView),
  NavigationActions: {
    reset() {
      return arguments
    },
    navigate() {
      return arguments
    },
  },
}))

jest.mock('react-native-firebase', () => ({
  notifications: jest.fn(() => ({
    onNotificationDisplayed: jest.fn(() => Promise.resolve()),
    onNotification: jest.fn(() => Promise.resolve()),
    onNotificationOpened: jest.fn(() => Promise.resolve()),
    getInitialNotification: jest.fn(() => Promise.resolve()),
  })),
  messaging: jest.fn(() => ({
    requestPermission: jest.fn(() => Promise.resolve()),
    onTokenRefresh: jest.fn(() => Promise.resolve()),
    getToken: jest.fn(() => Promise.resolve()),
  })),
}))

jest.mock('PixelRatio', () => ({
  get: jest.fn(() => 3), //PixelRatio.get() === 3 then iPhone 6 plus
  roundToNearestPixel: jest.fn(num => num),
}))

jest.mock('react-native-sensitive-info', () => {
  // todo: need to handle empty key-chain case
  const secureStorage = {
    ConnectMeKeyChain: {
      identifier: '26u37qFNUk3QwCUQ9ctest',
      phone: '6428900484',
      seed: 'S1StqtsAL4zkFZxmW57sL75zmufmVFQ3',
      pushComMethod: 'dehVox1KRqM:APA91bFWrJea1avml_ELw2MaH60abydtest',
    },
  }
  return {
    setItem: jest.fn((key, value, st) =>
      Promise.resolve((secureStorage['ConnectMeKeyChain'][key] = value))
    ),
    getItem: jest.fn(key =>
      Promise.resolve(secureStorage['ConnectMeKeyChain'][key])
    ),
    deleteItem: jest.fn(key =>
      Promise.resolve(delete secureStorage['ConnectMeKeyChain'][key])
    ),
  }
})

jest.mock('react-native-animatable', () => ({
  View: 'Animatable.View',
}))

jest.mock('react-native-camera', () => mockCamera)

jest.mock('Dimensions', () => ({
  get: jest.fn(() => ({
    width: 320,
    height: 540,
  })),
}))

jest.mock('react-native-touch-id', () => {
  return {
    authenticate: jest.fn(message => Promise.resolve()),
    isSupported: jest.fn(_ => Promise.resolve()),
  }
})

// jest.mock('react-native-sentry', () => ({
//   Sentry: {
//     install() {},
//     config() {
//       return { install() {} }
//     },
//     captureException(error) {},
//   },
//   SentryLog: {
//     Debug: 1,
//   },
// }))

jest.mock('AlertIOS', () => ({
  alert() {},
}))

global.fetch = fetch

jest.mock('Modal', () => mockModal)

jest.mock('react-native-modal', () => mockModal)

jest.mock('react-native-fetch-blob', () => ({
  fetch: jest.fn((type, url) => Promise.resolve()),
  fs: {
    dirs: {
      DocumentDir: '/var/application/DocumentDir',
    },
    exists: jest.fn(path => Promise.resolve(true)),
    unlink: jest.fn(path => Promise.resolve(true)),
    cp: jest.fn((source, destination) => Promise.resolve(destination)),
  },
}))

jest.mock('WebView', () => 'WebView')

jest.mock('InteractionManager', () => ({
  runAfterInteractions(fn) {
    fn()
  },
  createInteractionHandle: jest.fn(),
  clearInteractionHandle: jest.fn(),
}))

jest.mock('TextInput', () => 'TextInput')

jest.mock('moment', () =>
  jest.fn(date => ({
    format(format) {
      if (format === 'MM/DD/YYYY') {
        return '01/10/2018'
      }

      if (format === 'MMMM YYYY') {
        return 'January 2018'
      }

      return '2018-01-10T04:32:43+05:30'
    },
  }))
)

jest.mock('react-native-share', () => {})

jest.mock('react-native-branch', () => {
  return {
    ADD_TO_CART_EVENT: 'Add to Cart',
    ADD_TO_WISHLIST_EVENT: 'Add to Wishlist',
    PURCHASE_INITIATED_EVENT: 'Purchase Started',
    PURCHASED_EVENT: 'Purchased',
    REGISTER_VIEW_EVENT: 'View',
    SHARE_COMPLETED_EVENT: 'Share Completed',
    SHARE_INITIATED_EVENT: 'Share Started',

    STANDARD_EVENT_ADD_TO_CART: 'ADD_TO_CART',
    STANDARD_EVENT_ADD_TO_WISHLIST: 'ADD_TO_WISHLIST',
    STANDARD_EVENT_VIEW_CART: 'VIEW_CART',
    STANDARD_EVENT_INITIATE_PURCHASE: 'INITIATE_PURCHASE',
    STANDARD_EVENT_ADD_PAYMENT_INFO: 'ADD_PAYMENT_INFO',
    STANDARD_EVENT_PURCHASE: 'PURCHASE',
    STANDARD_EVENT_SPEND_CREDITS: 'SPEND_CREDITS',

    STANDARD_EVENT_SEARCH: 'SEARCH',
    STANDARD_EVENT_VIEW_ITEM: 'VIEW_ITEM',
    STANDARD_EVENT_VIEW_ITEMS: 'VIEW_ITEMS',
    STANDARD_EVENT_RATE: 'RATE',
    STANDARD_EVENT_SHARE: 'SHARE',

    STANDARD_EVENT_COMPLETE_REGISTRATION: 'COMPLETE_REGISTRATION',
    STANDARD_EVENT_COMPLETE_TUTORIAL: 'COMPLETE_TUTORIAL',
    STANDARD_EVENT_ACHIEVE_LEVEL: 'ACHIEVE_LEVEL',
    STANDARD_EVENT_UNLOCK_ACHIEVEMENT: 'UNLOCK_ACHIEVEMENT',

    // don't need any type checks for mocks, because they are not used
    // for type checking in code
    subscribe(cb: any) {},
  }
})

jest.mock('react-native-splash-screen', () => ({
  hide: jest.fn(),
}))

jest.mock('react-native-unique-id', () =>
  jest.fn(() => Promise.resolve('uniqueDeviceId'))
)

jest.mock('react-native-image-crop-picker', () => ({
  openPicker: jest.fn(() =>
    Promise.resolve({ path: '/var/application/DocumentDir/user-avatar.jpeg' })
  ),
}))

jest.mock('react-native-share', () => {})

// Add mock for RNIndy on NativeModules
// we don't want to mock NativeModules the way we are doing for all above
// because then we would have to mock everything else as well of NativeModules
// a lot of other modules depend on it, so we are adding RNIndy
// which is what we need to mock
NativeModules.RNIndy = {
  deserializeConnection: jest.fn(_ => Promise.resolve(1)),
  setWalletItem: jest.fn(_ => Promise.resolve(1)),
  deleteWalletItem: jest.fn(_ => Promise.resolve(1)),
  updateWalletItem: jest.fn(_ => Promise.resolve(1)),
  createWalletKey: jest.fn(_ => Promise.resolve('walletKey')),
  exitAppAndroid: jest.fn(),
}

jest.mock('react-native-version-number', () => ({
  appVersion: '1.0',
  buildVersion: '500',
}))

jest.mock('apptentive-react-native', () => ({
  Apptentive: {
    register: () => Promise.resolve(null),
    presentMessageCenter: jest.fn(),
    onAuthenticationFailed: reason => null,
  },
  ApptentiveConfiguration: () => null,
}))
jest.mock('BackHandler', () => ({
  addEventListener: jest.fn(),
  removeEventListener: jest.fn(),
}))
