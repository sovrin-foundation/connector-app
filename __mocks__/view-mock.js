// @flow
import React from 'react'
import { View } from 'react-native'

class MockView extends React.Component<any, void> {
  render() {
    return <View>{this.props.children}</View>
  }
}

export default MockView
