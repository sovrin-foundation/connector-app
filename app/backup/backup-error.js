// @flow

import React, { PureComponent } from 'react'
import { Image, StyleSheet, Dimensions } from 'react-native'
import { createStackNavigator } from 'react-navigation'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import {
  Container,
  CustomView,
  CustomText,
  CustomButton,
  Icon,
  CustomHeader,
} from '../components'
import {
  isBiggerThanShortDevice,
  grey,
  venetianRed,
  white,
} from '../common/styles/constant'
import { updateStatusBarTheme } from '../store/connections-store'
import { generateBackupFile } from '../backup/backup-store'
import { backupErrorRoute, exportBackupFileRoute } from '../common'
import type { BackupErrorProps } from './type-backup'
import headerStyles from './styles'
import {
  BACKUP_ERROR_BACK_TEST_ID,
  BACKUP_ERROR_CLOSE_TEST_ID,
} from './backup-constants'

const backImage = require('../images/icon_backArrow_white.png')
const closeImage = require('../images/iconClose.png')
export class BackupErrorScreen extends PureComponent<BackupErrorProps, void> {
  static navigationOptions = ({ navigation: { goBack, navigate, state } }) => ({
    header: (
      <CustomHeader backgroundColor={venetianRed} largeHeader flatHeader>
        <CustomView style={[headerStyles.headerSpacer]}>
          <Icon
            medium
            onPress={() => goBack(null)}
            testID={BACKUP_ERROR_BACK_TEST_ID}
            iconStyle={[headerStyles.headerBackIcon]}
            src={backImage}
          />
        </CustomView>

        <CustomView style={[headerStyles.headerSpacer]}>
          <Icon
            medium
            onPress={() => navigate(state.params.initialRoute)}
            testID={BACKUP_ERROR_CLOSE_TEST_ID}
            iconStyle={[headerStyles.headerIcon]}
            src={closeImage}
          />
        </CustomView>
      </CustomHeader>
    ),
    gesturesEnabled: true,
  })

  tryAgain = () => {
    const { initialRoute } = this.props.navigation.state.params
    this.props.generateBackupFile()
    this.props.navigation.navigate(exportBackupFileRoute, {
      initialRoute,
    })
  }

  render() {
    return (
      <Container
        safeArea
        testID={'restore-container'}
        style={[styles.backupErrorContainer]}
      >
        <Image
          source={require('../images/transparentBands2.png')}
          style={[styles.backgroundWaveImage]}
        />
        <Container testID={'restore-innercontainer'}>
          <CustomView center>
            <Image
              source={require('../images/alertRed.png')}
              style={[styles.errorImageContainer]}
            />
          </CustomView>

          <CustomView center pad>
            <CustomText
              transparentBg
              style={[styles.errorText]}
              center
              medium
              h4a
              center
            >
              There was a problem exporting your Connect.Me backup file. Please
              try again.
            </CustomText>
          </CustomView>
        </Container>
        <Container bottom pad>
          <CustomView pad>
            <CustomButton
              large={isBiggerThanShortDevice ? true : false}
              style={[styles.customButton, styles.buttonShadow]}
              title={'Try Again'}
              onPress={this.tryAgain}
              testID={'try-again'}
              customColor={{
                color: venetianRed,
                fontWeight: '600',
              }}
            />
          </CustomView>
        </Container>
      </Container>
    )
  }
}

const styles = StyleSheet.create({
  backupErrorContainer: {
    backgroundColor: venetianRed,
  },
  backgroundWaveImage: {
    position: 'absolute',
    top: isBiggerThanShortDevice ? '30%' : '20%',
    left: 0,
    right: 0,
    resizeMode: 'stretch',
    width: '100%',
  },
  customButton: {
    width: '100%',
    height: isBiggerThanShortDevice ? 57 : 43,
    borderRadius: 5,
    marginTop: 10,
  },
  buttonShadow: {
    shadowColor: grey,
    shadowOffset: {
      width: 0,
      height: 5,
    },
    shadowRadius: 5,
    shadowOpacity: 0.3,
    elevation: 4,
  },
  errorImageContainer: {
    marginTop: '5%',
    zIndex: -1,
  },
  errorText: {
    color: white,
  },
})

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      updateStatusBarTheme,
      generateBackupFile,
    },
    dispatch
  )

export default createStackNavigator({
  [backupErrorRoute]: {
    screen: connect(null, mapDispatchToProps)(BackupErrorScreen),
  },
})
