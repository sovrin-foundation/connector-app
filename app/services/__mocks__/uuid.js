// @flow
import { defaultUUID } from '../../../__mocks__/static-data'

export function uuid() {
  return defaultUUID
}
