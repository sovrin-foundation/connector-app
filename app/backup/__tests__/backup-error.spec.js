// @flow
import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { BackupErrorScreen } from '../backup-error'
import { getNavigation, getStore } from '../../../__mocks__/static-data'

describe('Restore screen', () => {
  function getProps() {
    return {
      store: getStore(),
      navigation: {
        navigate: jest.fn(),
        goBack: jest.fn(),
        state: {
          params: {
            recoveryPassphrase: '',
            initialRoute: '',
          },
        },
      },
      generateBackupFile: jest.fn(),
      updateStatusBarTheme: jest.fn(),
    }
  }

  function setup() {
    const props = getProps()
    return { props }
  }

  it('should render properly and match the snapshot', () => {
    const { props } = setup()
    const tree = renderer.create(<BackupErrorScreen {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  //TODO add snapshot for error screen
})
