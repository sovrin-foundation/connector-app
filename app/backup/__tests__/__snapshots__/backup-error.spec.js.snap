// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Restore screen should render properly and match the snapshot 1`] = `
<RCTSafeAreaView
  accessibilityLabel="restore-container"
  accessible={true}
  safeArea={true}
  style={
    Object {
      "backgroundColor": "#d0021b",
      "flex": 1,
    }
  }
  testID="restore-container"
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "left": 0,
          "position": "absolute",
          "resizeMode": "stretch",
          "right": 0,
          "top": "20%",
          "width": "100%",
        },
      ]
    }
  />
  <View
    accessibilityLabel="restore-innercontainer"
    accessible={true}
    style={
      Object {
        "flex": 1,
      }
    }
    testID="restore-innercontainer"
  >
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={1}
        style={
          Array [
            Object {
              "marginTop": "5%",
              "zIndex": -1,
            },
          ]
        }
      />
    </View>
    <View
      center={true}
      pad={true}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "paddingHorizontal": 15,
          "paddingVertical": 8,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 18,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#333333",
              "color": "#ffffff",
            },
            Object {
              "fontWeight": "500",
            },
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            null,
            Object {
              "color": "#ffffff",
            },
          ]
        }
      >
        There was a problem exporting your Connect.Me backup file. Please try again.
      </Text>
    </View>
  </View>
  <View
    pad={true}
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-end",
        "paddingHorizontal": 15,
        "paddingVertical": 8,
      }
    }
  >
    <View
      pad={true}
      style={
        Object {
          "paddingHorizontal": 15,
          "paddingVertical": 8,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "marginLeft": 15,
              "marginRight": 15,
            },
            undefined,
            Object {
              "marginLeft": 0,
              "marginRight": 0,
            },
            undefined,
          ]
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel="try-again"
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Array [
                Object {
                  "borderRadius": 5,
                  "height": 43,
                  "marginTop": 10,
                  "width": "100%",
                },
                Object {
                  "elevation": 4,
                  "shadowColor": "#777777",
                  "shadowOffset": Object {
                    "height": 5,
                    "width": 0,
                  },
                  "shadowOpacity": 0.3,
                  "shadowRadius": 5,
                },
              ],
            ]
          }
          testID="try-again"
          tvParallaxProperties={undefined}
        >
          <View
            pointerEvents="box-only"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#9E9E9E",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 19,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "backgroundColor": "#ffffff",
                },
                undefined,
                Object {
                  "padding": 12,
                },
                undefined,
                undefined,
                undefined,
                Array [
                  null,
                  Object {
                    "borderRadius": 5,
                    "height": 43,
                    "marginTop": 10,
                    "width": "100%",
                  },
                  Object {
                    "elevation": 4,
                    "shadowColor": "#777777",
                    "shadowOffset": Object {
                      "height": 5,
                      "width": 0,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                  },
                ],
                undefined,
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 16,
                    },
                    Object {
                      "color": "#d0021b",
                    },
                    Object {
                      "fontSize": 14,
                    },
                    undefined,
                    undefined,
                    Object {
                      "fontWeight": "600",
                    },
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Try Again
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
